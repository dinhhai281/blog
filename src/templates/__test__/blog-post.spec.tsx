import BlogPost, { BlogPostProps } from '@templates/blog-post';
import { layoutMockSetup, render } from '@utils/testing';
import React from 'react';

beforeAll(() => {
  layoutMockSetup();
});

describe('BlogPost Component', () => {
  let props: BlogPostProps;

  beforeEach(() => {
    props = {
      data: {
        markdownRemark: {
          frontmatter: {
            title: 'title',
            featuredImage: {
              childImageSharp: {
                fluid: {
                  src: 'src',
                  aspectRatio: 16 / 9,
                  sizes: 'sizes',
                  srcSet: 'srcSet',
                  base64: undefined,
                },
              },
            },
          },
          html: '<p>Test Blog Post</p>',
        },
      },
    };
  });
  it('render correctly', () => {
    const { asFragment } = render(<BlogPost data={props.data} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
