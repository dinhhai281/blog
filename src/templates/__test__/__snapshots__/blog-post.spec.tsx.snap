// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost Component render correctly 1`] = `
<DocumentFragment>
  <div
    class="container pt-12 mx-auto px-8 md:px-16 lg:px-32"
  >
    <header
      class="flex justify-between items-center mb-12"
    >
      <h1
        class="text-pink-700 text-6xl font-bold leading-10"
      >
        <a
          href="/en/"
        >
          title
        </a>
      </h1>
    </header>
    <main>
      <div
        class="pb-24"
      >
        <header>
          <h1
            class="mb-12 text-5xl text-pink-700"
          >
            title
          </h1>
          <div
            class="mb-8 mx-16 gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              aria-hidden="true"
              style="width: 100%; padding-bottom: 56.25%;"
            />
            <picture>
              <source
                sizes="sizes"
                srcset="srcSet"
              />
              <img
                alt=""
                loading="lazy"
                sizes="sizes"
                src="src"
                srcset="srcSet"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;source srcset="srcSet" sizes="sizes" /&gt;&lt;img loading="lazy" sizes="sizes" srcset="srcSet" src="src" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </header>
        <main>
          <p>
            Test Blog Post
          </p>
        </main>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
